jupyterhub/templates/scheduling/_scheduling-helpers.tpl

```diff
@@ -45,3 +56,103 @@ tolerations:
 {{- end }}
 {{- end }}
 {{- end }}
+
+
+
+{{- define "jupyterhub.nodeAffinityRequired" -}}
+{{- if .Values.singleuser.extraNodeAffinity.required -}}
+{{- .Values.singleuser.extraNodeAffinity.required | toYaml | trimSuffix "\n" | nindent 0 }}
+{{- end }}
+{{- end }}
+
+{{- define "jupyterhub.nodeAffinityPreferred" -}}
+{{- if .Values.singleuser.extraNodeAffinity.preferred -}}
+{{- .Values.singleuser.extraNodeAffinity.preferred | toYaml | trimSuffix "\n" | nindent 0 }}
+{{- end }}
+{{- end }}
+
+{{- define "jupyterhub.podAffinityRequired" -}}
+{{- if eq .podKind "core" -}}
+{{- else if eq .podKind "user" -}}
+{{- if .Values.singleuser.extraPodAffinity.required }}
+{{- .Values.singleuser.extraPodAffinity.required | toYaml | trimSuffix "\n" | nindent 0 }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{- define "jupyterhub.podAffinityPreferred" -}}
+{{- if eq .podKind "core" -}}
+{{- else if eq .podKind "user" -}}
+{{- if .Values.singleuser.extraPodAffinity.preferred -}}
+{{- .Values.singleuser.extraPodAffinity.preferred | toYaml | trimSuffix "\n" | nindent 0 }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{- define "jupyterhub.podAntiAffinityRequired" -}}
+{{- if eq .podKind "core" -}}
+{{- else if eq .podKind "user" -}}
+{{- if .Values.singleuser.extraPodAntiAffinity.required -}}
+{{- .Values.singleuser.extraPodAntiAffinity.required | toYaml | trimSuffix "\n" | nindent 0 }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{- define "jupyterhub.podAntiAffinityPreferred" -}}
+{{- if eq .podKind "core" -}}
+{{- else if eq .podKind "user" -}}
+{{- if .Values.singleuser.extraPodAntiAffinity.preferred -}}
+{{- .Values.singleuser.extraPodAntiAffinity.preferred | toYaml | trimSuffix "\n" | nindent 0 }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+
+
+{{- /*
+  input: podKind
+*/}}
+{{- define "jupyterhub.affinity" -}}
```

Comment:
This helper seems to be defined but unused

Reply:
Yes you are right, this PR only contained affinities required for configuring the user-related affinities, and this helper was not needed by the daemonset which only requires a subset of the affinity helpers introduced.